doc/developer/platform/ux.md

```diff
@@ -1,20 +1,726 @@
+Platform via the PostgreSQL protocol.
+
+Rationale: the PostgreSQL protocol does not have separate fields for an API
+client ID and API secret key, so both values need to be packed into the password
+field. Naively packing the UUIDs is unwieldy, however:
+
+```
+893677c2-a74c-4ecb-804d-3b7fa0c61fdc:68104da2-3417-442e-a614-f078dac470be
+```
+
+But since UUIDs are just 16 bytes each, we can pack them into a much shorter
+base64-encoded string:
+
+```py
+>>> base64.urlsafe_b64encode(client_id.bytes + secret_key.bytes).decode().rstrip('=')
+iTZ3wqdMTsuATTt_oMYf3GgQTaI0F0QuphTweNrEcL4
+```
+
+When encoded in this form, we call the value an app-specific password rather
+than an API key.
+
+In the web UI, we will need to build a custom interface for managing personal
+API tokens that shows the app-specific password form alongside the client ID
+and secret.
+
+### Billing
+
+Materialize Platform will bill monthly based on usage. Accounts will be billed
+in terms of compute units and storage units. A strawman proposal: compute units
+are machine-hours and storage units are gigabyte-months.
+
+We expect metering the compute usage of clusters and the storage usage of
+sources and tables to be straightforward. What is not clear is how to bill for
+the overhead of other system components.
+
+### Region
+
+When an account is created, it does not have any regions. The user's first step
+after registering for a new account is to choose one region to activate.
+
+Additional regions can be enabled via a "+ Enable new region" button in the
+region selector or somesuch in the web UI, or via the [global API](#global-api).
+
+To prevent disaster, regions cannot be deactivated. Users who wish to deactivate
+a region can contact support.
+
+### Cluster
+
+Clusters are created within a region via a new `CREATE CLUSTER` SQL statement:
+
+```sql
+CREATE CLUSTER <name> SIZE {xs|s|m|l|xl} [, REPLICATION FACTOR <n>];
+```
+
+Creating a cluster allocates persistent compute resources (e.g., EC2 instances)
+and begins billing the account for those resources. The SQL statement returns
+successfully as soon as the request is received.
+
+Specifying a replication factor greater than one will run multiple identical
+replicas of the cluster for redundancy. If one replica fails, requests can
+be routed to the other replica(s) until the failed replica is brought back
+online and brought up to date.
+
+A cluster may be horizontally scaled across multiple machines. The scaling is
+deliberately hidden from users behind the opaque notion of "size".
+
+The list of clusters and the status of each can be observed via the new `SHOW
+CLUSTERS` statement or the new `mz_clusters` system catalog table. The exact
+output of these commands is TODO, but they will include the provisioning status
+of the EC2 instances.
+
+It is possible to create indexes in a cluster before it is provisioned, but
+attempts to `SELECT` from these indexes will fail until the cluster is
+provisioned.
```

Comment:
Fail, or stall? I would have guessed stall, but if fail is important good to know!

Reply:
I'm not sure! I thought maybe fail was easier! Definitely a good thing to discuss from a "what is the desired UX" perspective?. I can imagine users wanting both. In the happy case, where a fresh cluster is booting up and will be available in 5s, you might just wanna wait. But if your cluster has fallen over and it's going to be 60m until it's caught up enough... maybe you'd rather an error message that tells you what's going on.

Reply:
This is definitely only at the level of "could discuss" which I think means "no need to solve to land this discussion foundation doc".

Reply:
Agree with the above. I guess this question is also related to Andy's question about "blocking" `CREATE CLUSTER` calls. If there's a way to easily wait for a cluster to be ready, we should be able to default to failing and let the user take care of waiting until things are ready.